client/my-sites/checkout/checkout-thank-you/jetpack-checkout-siteless-thank-you.tsx

```diff
@@ -157,41 +146,25 @@ const JetpackCheckoutSitelessThankYou: FC< Props > = ( {
 	}, [ calendlyUrl, currentUser, dispatch, productSlug ] );
 
 	// Update the ZD ticket linked to `receiptId` after the user has scheduled a call.
-	useEffect( () => {
-		const dispatchCalendlyEventScheduled = async ( event: CalendlyEvent ) => {
-			if ( event && event.data?.event === 'calendly.event_scheduled' ) {
-				const eventUri = event.data.payload?.event?.uri || '';
-				// The last part of the pathname is the ID of the Calendly event
-				const eventId = new URL( eventUri ).pathname.split( '/' ).pop();
-				const result = await addOnboardingCallInternalNote( receiptId, eventId );
-				if ( result ) {
-					dispatch(
-						recordTracksEvent( 'calypso_siteless_checkout_schedule_onboarding_call', {
-							product_slug: productSlug,
-							receipt_id: receiptId,
-							event_id: eventId,
-						} )
-					);
-				}
-			}
-		};
-
-		window.addEventListener( 'message', dispatchCalendlyEventScheduled );
-
-		return () => {
-			window.removeEventListener( 'message', dispatchCalendlyEventScheduled );
-		};
-	}, [] );
+	useSetCalendlyListenerEffect( { productSlug, receiptId, jetpackTemporarySiteId } );
 
 	useEffect( () => {
 		if ( supportTicketStatus === 'success' ) {
-			page( `/checkout/jetpack/thank-you-completed/no-site/${ productSlug }` );
+			const thankYouCompletedUrl = addQueryArgs(
+				{
+					jetpackTemporarySiteId,
+					productSlug,
```

Comment:
Is this necessary? The slug is already in the URL and it seems to be taken from it via route params (context.params.product) rather than the query.

Reply:
You might be correct. I'll review this again today.

Reply:
You were right. Thanks for catching this.